compositor_pipeline/src/pipeline/output/whip.rs

```diff
@@ -0,0 +1,246 @@
+            let elapsed_time = Instant::now().duration_since(start_time);
+            if elapsed_time > WHIP_INIT_TIMEOUT {
+                return Err(OutputInitError::WhipInitTimeout);
+            }
+            match init_confirmation_receiver.try_recv() {
+                Ok(result) => match result {
+                    Ok(_) => break,
+                    Err(err) => return Err(OutputInitError::WhipInitError(err.into())),
+                },
+                Err(err) => match err {
+                    oneshot::error::TryRecvError::Closed => {
+                        return Err(OutputInitError::UnknownWhipError)
+                    }
+                    oneshot::error::TryRecvError::Empty => {}
+                },
+            };
+        }
+
+        Ok(Self {
+            connection_options: options,
+            should_close,
+        })
+    }
+}
+
+impl Drop for WhipSender {
+    fn drop(&mut self) {
+        self.should_close
+            .store(true, std::sync::atomic::Ordering::Relaxed);
+    }
+}
+
+async fn run_whip_sender_task(
+    whip_ctx: WhipCtx,
+    packet_stream: PacketStream,
+    init_confirmation_sender: oneshot::Sender<Result<(), WhipError>>,
+    event_emitter: Arc<EventEmitter>,
+) {
+    let client = Arc::new(reqwest::Client::new());
+    let (peer_connection, video_track, audio_track) = match init_peer_connection().await {
+        Ok(pc) => pc,
+        Err(err) => {
+            init_confirmation_sender.send(Err(err)).unwrap();
+            return;
+        }
+    };
+    let output_id_clone = whip_ctx.output_id.clone();
+    let should_close_clone = whip_ctx.should_close.clone();
+    let whip_session_url = match connect(peer_connection, client.clone(), whip_ctx).await {
+        Ok(val) => val,
+        Err(err) => {
+            init_confirmation_sender.send(Err(err)).unwrap();
+            return;
+        }
+    };
+    init_confirmation_sender.send(Ok(())).unwrap();
+
+    for chunk in packet_stream {
+        if should_close_clone.load(std::sync::atomic::Ordering::Relaxed) {
+            break;
+        }
+        let chunk = match chunk {
+            Ok(chunk) => chunk,
+            Err(err) => {
+                error!("Failed to payload a packet: {}", err);
+                continue;
+            }
+        };
+
+        match chunk {
+            Payload::Video(video_payload) => match video_payload {
+                Ok(video_bytes) => {
+                    if video_track.write(&video_bytes).await.is_err() {
+                        error!("Error occurred while writing to video track for session");
+                    }
+                }
+                Err(err) => {
+                    error!("Error while reading video bytes: {err}");
+                }
+            },
+            Payload::Audio(audio_payload) => match audio_payload {
+                Ok(audio_bytes) => {
+                    if audio_track.write(&audio_bytes).await.is_err() {
```

Comment:
can we do something in case of error? Does it make sense to try next packet or maybe we should finish the output?